lib/src/map_mutations.dart

```diff
@@ -175,9 +176,18 @@ SourceEdit _removeFromBlockMap(
   final keySpan = keyNode.span;
   var end = getContentSensitiveEnd(valueNode);
   final yaml = yamlEdit.toString();
+  final lineEnding = getLineEnding(yaml);
```

Comment:
Side-note - we should probably cache the result of `getLineEnding` somewhere.
Not in this PR.

Reply:
yeah, some of this could benefit from more structure.
But performance isn't really much of a concern.